### DISCUSSION TITLE
Using the replacementName option makes Renovate to get the wrong digest if we are trying to use a different image

### DISCUSSION THREAD

**Discussion by ssegato81** [renovatebot/renovate#27728] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

ghcr.io/mend/renovate-ce:6.11.0-full

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

I'm using the replacementName option to suggest updating Docker Images to a different one because the original one is near the EOL.  

For example, I have an image called java8 (the version doesn't matter) and I want Renovate to create a PR updating that image to java11:2.0.0. I'm using a preset similar to the following one:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "configWarningReuseIssue": false,
  "extends": [
    "config:recommended"
  ],
  "packageRules": [
    {
      "matchDatasources": [
        "docker"
      ],
      "matchPackagePatterns": [
        "^artifacts.local.io/registry/java8.*"
      ],
      "replacementName": "artifacts.local.io/registry/java11",
      "replacementVersion": "2.0.0"
    }
  ]
}
```

In this case, Renovate tries to create the following PRs

1) PR that suggests updating java8 to a new minor/patch version
2) PR that suggests updating java8 to java11:2.0.0

Renovate fails to make the first PR (1) because it does the following

It iterates over three updates (bucket, replacement, and digest) but always replaces the packageName using the replacementName, so when it tries to get the digest for the java8:latest version (for example 1.0.2)  it uses the following information

packageName: java11 (it must be java8 because it's trying to figure out the digest associated to the latest java8 version)
newVersion: in this cases uses the java8 latest version

In shot, it creates a PR suggesting updating the java8 docker image version to the latest one using the  digest of the java11 docker image and java8 latest version.

There is an issue that has been created explaining this bug. I also added in more information to it. 

https://github.com/renovatebot/renovate/issues/20304

I've already fixed it locally and I have a PR that I want you to review it. 

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by ssegato81** [comment]
Link to a minimal reproduction: https://github.com/ssegato81/minimalReproduction

**Reply by viceice** [reply]
Will work on it today. Reproduction forked to <https://github.com/renovate-reproductions/27728-docker-replacement-digest>

**Comment by viceice** [comment, marked as answer]
- https://github.com/renovatebot/renovate/issues/20304
- https://github.com/renovatebot/renovate/pull/29164
